packages/components/src/components/editor/graph/edges/FloatingConnectionLine.tsx

```diff
@@ -0,0 +1,38 @@
+import { getCircleEdgeParams } from '@/utils/graph';
+import { getBezierPath, type ConnectionLineComponentProps, type InternalNode } from '@xyflow/react';
+
+function FloatingConnectionLine({ toX, toY, fromPosition, toPosition, fromNode }: ConnectionLineComponentProps) {
+  if (!fromNode) {
+    return null;
+  }
+
+  const targetNode = {
+    id: 'connection-target',
+    measured: {
+      width: 1,
+      height: 1
+    },
+    internals: {
+      positionAbsolute: { x: toX, y: toY }
+    }
+  };
+
+  const { sx, sy } = getCircleEdgeParams(fromNode, targetNode as InternalNode);

```

Comment:
instead of hard cast here and trick typescript, better define the type at the root and even help typescript define this object for you `const targetNode: InternalNode = {};`